Services/Skill/classes/class.ilSkillProfile.php

```diff
@@ -567,21 +573,31 @@ public function getAssignedRoles()
     {
         $ilDB = $this->db;
         $lng = $this->lng;
+        $review = $this->review;
 
         $set = $ilDB->query(
             "SELECT * FROM skl_profile_role " .
             " WHERE profile_id = " . $ilDB->quote($this->getId(), "integer")
         );
         $roles = array();
         while ($rec = $ilDB->fetchAssoc($set)) {
-            $name = ilObjRole::_lookupTitle($rec["role_id"]);
+            $name = ilObjRole::_getTranslation(ilObjRole::_lookupTitle($rec["role_id"]));
             $type = $lng->txt("role");
+            // get object of role
+            $obj = ilContainerReference::_lookupObjectId($review->getObjectReferenceOfRole($rec["role_id"]));
```

Comment:
This looks suspicious to me. Please ask Stefan if this is the right way to do this.

Reply:
ilContainerReference ist die eine Basisklasse für alle Referenzobjekte (Kurs-, Gruppen, Kategorienverknüpfung)
DIese Klasse implementiert die Methode "_lookupObjId()" nicht, sondern erbt sie von ilObject. 
Dein Code funktioniert, ich würde aber ilContainerReference::_lookupObjectId durch ilObject::_lookupObjectId
ersetzen.

Reply:
Danke! Keine Ahnung mehr, warum ich da unbedingt ilContainerReference benutzen wollte...